zerver/models.py

```diff
@@ -2241,6 +2241,7 @@ class PasswordTooWeakError(Exception):
 
 class UserGroup(models.Model):  # type: ignore[django-manager-missing] # django-stubs cannot resolve the custom CTEManager yet https://github.com/typeddjango/django-stubs/issues/1023
     MAX_NAME_LENGTH = 100
+    INVALID_NAME_PREFIXES = ["@", "role:", "user:", "stream:"]
```

Comment:
Let's add `channel:` to this list for forward-safety if we rename `stream` to `channel` at some point.